C4G.Unity/Assets/C4G/Editor/Scripts/C4GSettings.cs

```diff
@@ -38,7 +45,44 @@ public string SerializedConfigsFolderFullPath
 
         public bool IsGeneratedCodeFolderValid => !string.IsNullOrEmpty(_generatedCodeFolderPath) &&
                                                   Directory.Exists(GeneratedCodeFolderFullPath);
+
         public bool IsSerializedConfigsFolderValid => !string.IsNullOrEmpty(_generatedCodeFolderPath) &&
-                                                  Directory.Exists(SerializedConfigsFolderFullPath);
+                                                      Directory.Exists(SerializedConfigsFolderFullPath);
+
+        public IReadOnlyList<string> SheetNames => _sheetNames;
+
+        private readonly List<IAliasDefinition> _aliasDefinitionsConvertBuffer = new List<IAliasDefinition>();
+        public IReadOnlyList<IAliasDefinition> AliasDefinitions
+        {
+            get
+            {
+                _aliasDefinitionsConvertBuffer.Clear();
```

Comment:
[Critical] Let's use `dirty` flag here to avoid expensive reflection and list recreation on every `get`.

smth like,
```csharp
private List<IAliasDefinition> _cachedAliasDefinitions;
private bool _aliasDefinitionsCacheDirty = true;

public IReadOnlyList<IAliasDefinition> AliasDefinitions
{
    get
    {
        if (_aliasDefinitionsCacheDirty)
        {
            RebuildAliasDefinitionsCache();
            _aliasDefinitionsCacheDirty = false;
        }
        return _cachedAliasDefinitions;
    }
}

private void OnValidate()
{
    _aliasDefinitionsCacheDirty = true;
}
```